### ISSUE TITLE
execution order of transform.Compose

### ISSUE THREAD

**Issue by mryzlee** [2017-fall-DL-training-program/ConvNetwork#1]
Is it "In-Order" execution in transform.Compose?

If yes, the Normalize will be the last step in the following example codes:
transform_train = transforms.Compose([
    transforms.RandomCrop(32, padding=4),
    transforms.RandomHorizontalFlip(),
    transforms.ToTensor(),
    transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
])

However, it also means that the zero padding pixels will affect the normalized values of other real image pixels.
Besides, zero padding will generates extra horizontal edge/gradient and vertical edge/gradient between real image pixels and zero padding pixels.
Is there any better padding method such as boundary extension to avoid this problem in pyTorch?

**Comment by JiaRenChang** [comment]
Hi,
   The means and standard deviations are computed from full training set in the Cifar-10.
   Any pictures that we want to train/test should be normalized to this space (including padded or flipped images).
   PyTorch only offers zero padding inside RandomCrop() because it is fast and simple.

Jia-Ren
